pkg/cmd/infra/router/router.go

```diff
@@ -37,35 +54,75 @@ type templateRouterConfig struct {
 	StatsUsername      string
 }
 
+// f5RouterConfig is the config necessary to start an F5 router plugin
+type f5RouterConfig struct {
+	Host     string
+	Username string
+	Password string
+	Vserver  string
+	Insecure bool
+}
+
+const (
+	//routerTypeF5 is the type value for an f5 router plugin
+	routerTypeF5 = "f5"
+	//routerTypeTemplate is the type value for a template router plugin
+	routerTypeTemplate = "template"
+)
+
+func newEmptyConfig() *routerConfig {
+	return &routerConfig{
+		Config:               clientcmd.NewConfig(),
+		TemplateRouterConfig: &templateRouterConfig{},
+		F5RouterConfig:       &f5RouterConfig{},
+	}
+}
+
 // NewCommndTemplateRouter provides CLI handler for the template router backend
 func NewCommandTemplateRouter(name string) *cobra.Command {
-	cfg := &templateRouterConfig{
-		Config: clientcmd.NewConfig(),
-	}
+	cfg := newEmptyConfig()
 
 	cmd := &cobra.Command{
 		Use:   fmt.Sprintf("%s%s", name, clientcmd.ConfigSyntax),
 		Short: "Start an OpenShift router",
 		Long:  routerLong,
 		Run: func(c *cobra.Command, args []string) {
+			if !isValidType(cfg.Type) {
+				glog.Fatal("Invalid type specified.  Valid values are template, f5")
+			}
+
 			defaultCert := util.Env("DEFAULT_CERTIFICATE", "")
 			if len(defaultCert) > 0 {
-				cfg.DefaultCertificate = defaultCert
+				cfg.TemplateRouterConfig.DefaultCertificate = defaultCert
 			}
 
 			routerSvcNamespace := util.Env("ROUTER_SERVICE_NAMESPACE", "")
 			routerSvcName := util.Env("ROUTER_SERVICE_NAME", "")
-			cfg.RouterService = ktypes.NamespacedName{
+			cfg.TemplateRouterConfig.RouterService = ktypes.NamespacedName{
 				Namespace: routerSvcNamespace,
 				Name:      routerSvcName,
 			}
 
-			plugin, err := makeTemplatePlugin(cfg)
+			var plugin router.Plugin
+			var err error
+
+			switch cfg.Type {
+			case "f5":
+				plugin, err = makeF5Plugin(cfg)
+			case "template":
+				plugin, err = makeTemplatePlugin(cfg)
+			}
```

Comment:
OK, I'll add a `default:` case that assigns a non-nil error message to `err` for the `if` after the `switch`.